wifiphisher/common/deauth.py

```diff
@@ -42,39 +44,54 @@ def __init__(self, ap_bssid, jamming_interface):
         self._ap_bssid = ap_bssid
         self._should_continue = True
         self._jamming_interface = jamming_interface
+        self._ap_iface_mac = ap_iface_mac
         self._non_client_addresses = constants.NON_CLIENT_ADDRESSES
-        self._send_deauth_packets_thread = threading.Thread(\
-                target = self._send_deauthentication_packets)
-        self._find_client_thread = threading.Thread(\
-                target=self._find_clients)
+        self._deauth_thread = None
+        self._find_client_thread = None
 
         # create a socket for sending packets
         self._socket = linux.L2Socket(iface=self._jamming_interface)
 
         # craft and add deauthentication packet to broadcast address
-        self._craft_and_add_packet(self._ap_bssid, constants.WIFI_BROADCAST)
+        # add broadcast deauth when --essid not being used
+        if not self._is_frenzy_deauth():
```

Comment:
@blackHatMonkey thank you for your review. I'll modify this part. I remembered you have told me once about this discipline, and just want to make sure I understand correctly. Today I have tried to write the test cases for the `NetworkManager` and I found that since in the `__init__` of `NetworkManager`  do some operations [related to pyw module] to init the `_interfaces` dictionary, and this makes me have to use `MagicMock` to make a fake object for our `NetworkManager` which is inconvenient. Is this the reason not to do operations in `__init__` function?